services/datamanager/builtin/capture_test.go

```diff
@@ -235,13 +198,126 @@ func TestDataCaptureEnabled(t *testing.T) {
 	}
 }
 
-func waitForCaptureFiles(captureDir string) {
+func TestSwitchResource(t *testing.T) {
+	captureDir := t.TempDir()
+	mockClock := clk.NewMock()
+	// Make mockClock the package level clock used by the dmsvc so that we can simulate time's passage
+	clock = mockClock
+
+	// Set up robot config.
+	config, deps := setupConfig(t, enabledTabularCollectorConfigPath)
+	config.CaptureDisabled = false
+	config.ScheduledSyncDisabled = true
+	config.CaptureDir = captureDir
+
+	// Build and start data manager.
+	dmsvc, r := newTestDataManager(t)
+	defer func() {
+		test.That(t, dmsvc.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	resources := resourcesFromDeps(t, r, deps)
+	err := dmsvc.Reconfigure(context.Background(), resources, resource.Config{
+		ConvertedAttributes: config,
+	})
+	test.That(t, err, test.ShouldBeNil)
+	passTimeCtx1, cancelPassTime1 := context.WithCancel(context.Background())
+	donePassingTime1 := passTime(passTimeCtx1, mockClock, captureInterval)
```

Comment:
Why do we need to pass time more than a single capture interval here?

Reply:
The first passTime is to make sure that the initial resource is capturing data, then the resource is changed, and the second passTime is to test that the collector is capturing data as expected after that change.

Reply:
Ah I see - the format of the previously existing tests suggests that we need more than one iteration until the sensor data shows up. When I add only one interval to the mock clock (instead of calling passTime) the test fails bc there isn't sensor data yet. I don't have enough context on data capture to say why there's this lag, but since this behavior is present in the existing tests I think this should be okay!

Reply:
Ah, that's because of buffered IO. Ok that's cool then